aten/src/ATen/SparseCsrTensorUtils.h

```diff
@@ -366,5 +367,37 @@ inline bool only_sparse_compressed_add_trivial_cases(
       });
 }
 
+inline Tensor to_type(Tensor input, ScalarType dtype) {
+  Tensor compressed_indices, plain_indices;
+  std::tie(compressed_indices, plain_indices) =
+      at::sparse_csr::getCompressedPlainIndices(input);
+  return at::_sparse_compressed_tensor_unsafe(
+      std::move(compressed_indices),
+      std::move(plain_indices),
+      std::move(input.values()).to(dtype),
+      input.sizes(),
+      dtype,
+      input.layout(),
+      input.device(),
+      input.options().pinned_memory_opt());
+}
+
+template <typename acc_t, typename scalar_t>
+inline void create_acc_buffer(
```

Comment:
I think we can also include the construction and resize of `Tensor new_values, new_values_acc;` into this helper function.